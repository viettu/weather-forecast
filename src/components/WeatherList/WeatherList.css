.WeatherList-container {
    list-style: none;
    margin: 0;
    padding: 0;

    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

.WeatherList-item {
    margin: 0 10px;
}
