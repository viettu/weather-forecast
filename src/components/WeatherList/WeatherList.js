import React from 'react'
import PropTypes from 'prop-types'

import './WeatherList.css'

import WeatherItem from '../WeatherItem/WeatherItem'

const WeatherList = ({weathers}) => {
    const list = (weathers || []).map((weather) => {
        return (
            <li key={weather.id} className="WeatherList-item">
                <WeatherItem
                    applicableDate={weather.applicableDate}
                    maxTemp={weather.maxTemp}
                    minTemp={weather.minTemp}
                />
            </li>
        )
    })

    return <ul className="WeatherList-container">{list}</ul>
}

WeatherList.propTypes = {
    weathers: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.number,
            applicableDate: PropTypes.string,
            minTemp: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
            maxTemp: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        })
    ),
}

export default WeatherList
