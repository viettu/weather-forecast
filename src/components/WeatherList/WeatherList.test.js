import React from 'react';
import {render, screen, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';

import WeatherList from './WeatherList';

afterEach(() => {
    cleanup();
});

test('renders without crashing', () => {
    render(<WeatherList />);
});

test('snapshot testing', () => {
    let renderResult = render(<WeatherList />);
    expect(renderResult.container.firstChild).toMatchSnapshot();

    const testData = [
        {
            id: 1,
            applicableDate: '2020-10-28',
            minTemp: 23.97,
            maxTemp: 29.259999999999998,
        },
    ];

    renderResult = render(<WeatherList weathers={testData} />);
    expect(renderResult.container.firstChild).toMatchSnapshot();
});

test('should display an array of WeatherItem', () => {
    const testData = Array(5)
        .fill(0)
        .map((_, idx) => ({
            id: idx,
            applicableDate: '2020-10-28',
            minTemp: 23.97,
            maxTemp: 29.259999999999998,
        }));

    render(<WeatherList weathers={testData} />);

    expect(screen.queryAllByText('Wednesday').length).toBe(5);
    expect(screen.queryAllByText('24.0 - 29.3 â„ƒ').length).toBe(5);
});
