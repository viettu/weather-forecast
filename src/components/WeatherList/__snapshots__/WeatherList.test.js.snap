// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<ul
  class="WeatherList-container"
/>
`;

exports[`snapshot testing 2`] = `
<ul
  class="WeatherList-container"
>
  <li
    class="WeatherList-item"
  >
    <div
      class="WeatherItem-container"
    >
      <h3>
        Wednesday
      </h3>
      <div
        class="WeatherItem-temperature"
      >
        24.0 - 29.3 â„ƒ
      </div>
    </div>
  </li>
</ul>
`;
